src/widgets/index/index.ts

```diff
@@ -434,6 +469,35 @@ const index = (props: IndexProps): Index => {
         helper!.lastResults = results;
       });
 
+      localWidgets.forEach(widget => {
```

Comment:
You can merge this loop with the following one.

Reply:
Nope, otherwise we won't get the full render state passed to the widget when they `render`. The goal of having this as a loop before rendering is to do a full pass to construct the render state.

Reply:
Ok, since this particularity is not explicit in the code could you add a comment to explain it?

Reply:
b84947b56f2238a4c5a3f131ce4c9205f94cae0c